src/scheduled_tasks.rs

```diff
@@ -0,0 +1,83 @@
+// Scheduler, and trait for .seconds(), .minutes(), etc.
+use clokwerk::{Scheduler, TimeUnits};
+// Import week days and WeekDay
+use diesel::{sql_query, PgConnection, RunQueryDsl};
+use lemmy_db_queries::{source::activity::Activity_, DbPool};
+use lemmy_db_schema::source::activity::Activity;
+use log::info;
+use std::{thread, time::Duration};
+
+/// Schedules various cleanup tasks for lemmy in a background thread
+pub fn setup(pool: DbPool) {
+  let mut scheduler = Scheduler::new();
+
+  // Reindex the aggregates tables every one hour
+  // This is necessary because hot_rank is actually a mutable function:
+  // https://dba.stackexchange.com/questions/284052/how-to-create-an-index-based-on-a-time-based-function-in-postgres?noredirect=1#comment555727_284052
```

Comment:
Read this if you want to see some DBA's hate on me :) . I can't find another way around this issue tho. The performance gain from adding an index (even one that degrades over time), is gigantic: (2 seconds vs 20 ms).